pkg/controller/pravegacluster/upgrade.go

```diff
@@ -404,6 +405,139 @@ func (r *ReconcilePravegaCluster) syncSegmentStoreVersion(p *pravegav1alpha1.Pra
 	return false, nil
 }
 
+//this function is to check are we doing a rollback in case of a upgrade failure while upgrading from a version below 07 to a version above 07
+func (r *ReconcilePravegaCluster) IsClusterRollbackingFrom07(p *pravegav1alpha1.PravegaCluster) bool {
+	if !r.isRollbackTriggered(p) {
+		return false
+	}
+	//checking if sts above07 exsists
+	stsAbove07 := &appsv1.StatefulSet{}
+	name := util.StatefulSetNameForSegmentstoreAbove07(p.Name)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, stsAbove07)
+	if err != nil && errors.IsNotFound(err) {
+		return false
+	}
+	//checking if sts below07 exsists
+	stsBelow07 := &appsv1.StatefulSet{}
+	name = util.StatefulSetNameForSegmentstoreBelow07(p.Name)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, stsBelow07)
+	if err != nil && errors.IsNotFound(err) {
+		return false
+	}
+	//true will only be return if both sts are present
+	return true
+}
+
+//this function calls the syncstoreversion for upgrade and rollback operation
+func (r *ReconcilePravegaCluster) syncStoreVersion(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	//in case of a upgrade from a version below 07 to a version above 07 syncSegmentStoreVersionTo07() is called and the util.IsClusterUpgradingTo07(p)  is true in this case
```

Comment:
Please remove these 2 comments they are unnecessary, code is self explanatory

Reply:
done